compiler/src/iree/compiler/Codegen/Common/TransformExtensions/CommonExtensionsOps.td

```diff
@@ -82,4 +83,40 @@ def IREEBufferizeOp : Op<Transform_Dialect, "iree.bufferize",
   let cppNamespace = "mlir::iree_compiler::IREE::transform_dialect";
 }
 
+def MatchInterfaceEnum : I32EnumAttr<"MatchInterfaceEnum", "An interface to match",
+    [
+      I32EnumAttrCase<"LinalgOp", 0>,
+      I32EnumAttrCase<"TilingInterface", 1>
+    ]>{
+  let cppNamespace = "::mlir::iree_compiler::IREE::transform_dialect";
+}
+
+def MatchOp : Op<Transform_Dialect, "iree.match",
+    [FunctionalStyleTransformOpTrait, 
+     MemoryEffectsOpInterface,
+     DeclareOpInterfaceMethods<TransformOpInterface>]> {
+  let description = [{
+    Match op with the specified constraints, within the target op.
+
+    The following constraints are supported:
+      - interface: an optional MatchInterfaceEnum specifying an enume
+      representation for an interface to target.
+      - ops: an optional StrArrayAttr specifying the concrete name of an op. Note: 
+      either name or interface must be specified.
+  }];
+
+  let arguments = (ins PDL_Operation:$target,
+                       OptionalAttr<StrArrayAttr>:$match_op,
+                       OptionalAttr<MatchInterfaceEnum>:$match_interface);
```

Comment:
To make the list of supported interfaces easier to extend, this could be a string. Then just the .cpp file needs extending; could even be done with 1-liner macro. On the other hand, this could give the false impression to users that all interfaces are supported (when just looking at the IR).

Reply:
Right, I'd like to avoid this false sense of "it just works for all interfaces". However if over time usage overwhelmingly supports a string based impl we can shift to that when the time comes.